spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/cql/QueryOptionsUtil.java

```diff
@@ -126,6 +129,9 @@ public static Update addWriteOptions(Update update, WriteOptions writeOptions) {
 		if (!writeOptions.getTtl().isNegative()) {
 			update.using(QueryBuilder.ttl(Math.toIntExact(writeOptions.getTtl().getSeconds())));
 		}
+		if (!(writeOptions.getTimestamp() < 0)) {
```

Comment:
`WriteOptions` (and `QueryOptions` in general) uses nullable properties to indicate presence/absence of options.